linked_list/linked_list.py

```diff
@@ -23,13 +25,55 @@ def __init__(self):
     self.tail = None
 
   def add_to_tail(self, value):
-    pass
 
+    new_node = Node(value)
+    if self.head == None:
+        self.head = new_node
+    else:
+      self.tail.set_next(new_node)
+    self.tail = new_node
+ 
   def remove_head(self):
-    pass
+
+    if self.head == None:
+      return None
+
+    if self.head.get_next() == None:
+      removed = self.head.get_value()
+      self.head = None
+      self.tail = None
+
+    elif self.head is not None:
+      removed = self.head.get_value()
+      current = self.head.get_next()
+      self.head = current
+
+    return removed
 
   def contains(self, value):
-    pass
+    current = self.head
+    found = False
+    while current and found is False:
+      if current.get_value() == value:
+        found = True
+      else:
+        current = current.get_next()
+    if current is None:
+      return None
+    return current
 
   def get_max(self):
-    pass
+    if self.head == None or self.tail == None:
+      return None
+    current = self.head
+    max_val = -math.inf
+
+    while current.next_node != None:
+      if current.get_value() > max_val:
+        max_val = current.get_value()
+      current = current.get_next()
+
+    if self.tail.get_value() > max_val:
+      max_val = self.tail.get_value()
+    
```

Comment:
Very minor tip: if you just simply check if `current` is truthy, or more explicitly `current != None`, you wouldn't have to do a `self.tail` check in the end:
```
  def get_max(self):
    if self.head == None or self.tail == None:
      return None
    current = self.head
    max_val = -math.inf

    while current != None:
      if current.get_value() > max_val:
        max_val = current.get_value()
      current = current.get_next()
    
    return max_val
```

Reply:
ok, thanks!